talentprotocol/mvp
# Ability to disable messages

## Summary




### What changed?

- I've added a flag (`messaging_disabled`) that turns off messaging. 
- Going forward it shouldn't be possible to message users with messaging disabled. 
- Users with messaging disabled are not able to send messages either.
- Message buttons should not be displayed when users have messaging disabled.

### Why it changed?

Users might choose to receive messages or not.

### How it changed?

I've added the `messaging_disabled` flag to the users table. Defaulted to `false` which means that messaging will be enabled by default. 

## Notion link

https://www.notion.so/talentprotocol/Improve-messages-85e472e207af4105a3279ea02fb04171#f5c58b96799e4346bef872a978f520d8

## How to test?

1. Have two sessions with different users
2. Enable messages in both and send message (Assert it sends)
3. Disable messaging in one of the users and assert both users can send messages in their conversation
4. Switch the flags in both users and assert both users can send messages in their conversation
5. Enable messaging in both and assert message are now sent
6. Ensure message button is hidden on the supporters list if the supporter does not have messaging enable

## Notes

<img width="1662" alt="Screenshot 2022-02-13 at 23 09 34" src="https://user-images.githubusercontent.com/8901722/153779730-886fe91c-3530-4daf-91d6-cb9db5e7b7e5.png">
<img width="1920" alt="Screenshot 2022-02-13 at 23 09 50" src="https://user-images.githubusercontent.com/8901722/153779733-101d1ba0-2c93-4f3c-a658-1686221663f2.png">
<img width="1920" alt="Screenshot 2022-02-13 at 23 09 58" src="https://user-images.githubusercontent.com/8901722/153779735-d77fbd1e-43de-410d-a67f-d28a0c2cc949.png">
